src/hw5/COMP232LinkedBinarySearchTree.java

```diff
@@ -170,8 +194,37 @@ private void addNodeToSubTree(BTNode<K, V> subTreeRoot,
 	 * {@inheritDoc}
 	 */
 	public V remove(K key) {
```

Comment:
This function does not fully remove the nodes, for instance in testRemoveRightSubChildLeaf we expect the in order traversal to be "ABDEFG" while yours returned "ABCDEFG", so the "C" node remained. In this example, we should test if "C" is a leaf node and see if it is the left or right child of its parent, and then assign cur.parent.left (or right if it is a right child) to be null.